lib/ledger-node.js

```diff
@@ -18,18 +18,25 @@ module.exports = class LedgerNode {
     this.id = options.id;
     this.ledger = options.ledger;
     this.consensus = options.consensus;
-    this.meta = new LedgerNodeMeta(options.storage.meta, options.consensus);
-    this.blocks = new LedgerNodeBlocks(
-      this, options.storage, options.consensus);
-    this.config = new LedgerNodeConfig(
-      this, options.storage, options.consensus);
-    this.events = new LedgerNodeEvents(
-      this, options.storage, options.consensus);
-    this.operations = new LedgerNodeOperations(
-      this, options.storage, options.consensus);
-    this.stateMachine = new LedgerNodeStateMachine(
-      this, options.storage, options.consensus);
     this.storage = options.storage;
     this.driver = {};
+    const ledgerNode = this;
+    this.meta = new LedgerNodeMeta({ledgerNode});
+    // this.meta = new LedgerNodeMeta(options.storage.meta, options.consensus);
+    this.blocks = new LedgerNodeBlocks({ledgerNode});
+    // this.blocks = new LedgerNodeBlocks(
+    //   this, options.storage, options.consensus);
+    this.config = new LedgerNodeConfig({ledgerNode});
+    // this.config = new LedgerNodeConfig(
+    //   this, options.storage, options.consensus);
+    this.events = new LedgerNodeEvents({ledgerNode});
+    // this.events = new LedgerNodeEvents(
+    //   this, options.storage, options.consensus);
+    this.operations = new LedgerNodeOperations({ledgerNode});
+    // this.operations = new LedgerNodeOperations(
+    //   this, options.storage, options.consensus);
+    this.stateMachine = new LedgerNodeStateMachine({ledgerNode});
```

Comment:
I feel like this is a weird name and it could be like `state` or `records` or something. Thoughts?

Reply:
A name that make sense to me is something like `recordState`.

or maybe it could be `ledgerNode.records.getState({recordId})`.

@dlongley do you have some conception of other APIs that will be added here?

Reply:
I think `ledgerNode.records.get({recordId, maxBlockHeight})` makes the most sense to me. I don't think we need to tack "state" on there if we're using "records". The parameters allow you to select a record in a past state. No other APIs planned at this time ... maybe some kind of query ... but that might just be a different option on `get`.